File: src/Scripting/Lua/patterns/microkernel.lua
local kernel = { plugins = {} }
function kernel:register(name, plugin) self.plugins[name] = plugin end
function kernel:execute(name, value) return self.plugins[name](value) end
kernel:register("double", function(x) return x * 2 end)
assert(kernel:execute("double", 6) == 12)
return true

File: src/Scripting/PHP/patterns/microkernel.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('microkernel');
$plugins = [];
$register = static function(string $name, callable $plugin) use (&$plugins): void { $plugins[$name] = $plugin; };
$run = static fn(string $name, mixed $input) => $plugins[$name]($input);
$register('upper', 'strtoupper');
$check($run('upper','kernel') === 'KERNEL');
